src/borg/archiver.py

```diff
@@ -1517,6 +1521,53 @@ def do_prune(self, args, repository, manifest, key):
             if num is not None:
                 keep += prune_split(archives, rule, num, kept_because)
 
+        if args.filter_stdio:
+            format = "{archive:<36} {time} [{id}]{NL}"
+            formatter = ArchiveFormatter(format, repository, manifest, key, json=True, iec=args.iec)
+            output_data = []
+            for archive in archives:
+                output_data.append(formatter.get_item_data(archive))
+            json_print(basic_json_data(manifest, extra={'archives': output_data}))
+
+            def hex_to_bin(d):
+                bin_id = None
+                try:
+                    bin_id = unhexlify(d)
+                except:
+                    raise ValueError('Invalid value, must be 64 hex digits') from None
+                if len(bin_id) != 32:
+                    raise ValueError('Invalid value, must be 64 hex digits')
+                return bin_id
+
+            archives_by_id = {archive.id: archive for archive in archives}
+            archives_by_name = {archive.name: archive for archive in archives}
+
+            kept_counter = 0
+            try:
+                input = sys.stdin.read()
+                input_data = []
+                if input:
+                    input_data = json.loads(input)
```

Comment:
i think we must require the input to be always valid json (even if it is just `[]`) and reject everything else, including the empty (`''`) input.

Reply:
See abov; will change...